src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeExperiment.groovy

```diff
@@ -14,239 +16,258 @@ import java.lang.reflect.Method
+    return new OxfordNanoporeExperiment(sampleId, measurements)
+  }
+
+  /**
+   * Provides a list of measurements contained within the experiment folder.
+   * @return
+   */
+  List<OxfordNanoporeMeasurement> getMeasurements() {
+    return this.measurements
+  }
+
+  @Override
+  String getSampleCode() {
+    return this.sampleId
+  }
+
+  /**
+   * Helper method that parses the QBiC identifier from the root folder name
+   */
+  private static String parseQbicIdFromRootFolder(Map nanoPoreSequencerOutput) {
+    def name = Objects.requireNonNull(nanoPoreSequencerOutput.get("name"), "The root folder must contain a name property.")
+    final def ids = SampleCodeFunctions.findAllQbicSampleCodes(name as String)
+    if (ids.isEmpty()) {
+      throw new IllegalArgumentException("No QBiC sample identifier found!")
+    }
+    if (ids.size() > 1) {
+      throw new IllegalArgumentException("Name contained more than one valid sample id!")
     }
+    return ids.get(0)
+  }
 
-    @Override
-    String getSampleCode() {
-        return this.sampleId
+  /**
+   * Helper method that creates the measurements from the sequencer output
+   */
+  private static List<OxfordNanoporeMeasurement> parseMeasurements(Map nanoPoreSequencerOutput) {
+    final def measurements = []
+    Objects.requireNonNull(nanoPoreSequencerOutput.get("children"), "The root folder must contain at least one measurement folder.")
+    nanoPoreSequencerOutput.get("children").each { Map measurementItem ->
+      def name = measurementItem.get("name") as String
+      def relativePath = measurementItem.get("path") as String
+      def children = parseMeasurementItems(measurementItem.get("children") as List)
+      def metadata = measurementItem.get("metadata") as Map
+      measurements.add(new OxfordNanoporeMeasurement(name, relativePath, children, metadata))
     }
+    return measurements
+  }
 
-    /**
-     * Helper method that parses the QBiC identifier from the root folder name
-     */
-    private static String parseQbicIdFromRootFolder(Map nanoPoreSequencerOutput) {
-        def name = Objects.requireNonNull(nanoPoreSequencerOutput.get("name"), "The root folder must contain a name property.")
-        final def ids = SampleCodeFunctions.findAllQbicSampleCodes(name as String)
-        if (ids.isEmpty()) {
-            throw new IllegalArgumentException("No QBiC sample identifier found!")
+  /*
+   * Helper method that creates a list of mixed DataFolders and DataFiles instances
+   */
+
+  private static List<?> parseMeasurementItems(List<Map> items) {
+    final def children = []
+    items.each { item ->
+      {
+        if (item.get(FILEKEY)) {
+          // Lets try to parse it as a subclass of a DataFile
+          DataFile putativeFile = parseFile(item)
+          children.add(putativeFile)
         }
-        if (ids.size() > 1) {
-            throw new IllegalArgumentException("Name contained more than one valid sample id!")
+        if (item.get(FOLDERKEY)) {
```

Comment:
Same as with the `FILEKEY`, I think this is not very easy to read and understand.

Reply:
It's adapted nice idea 👍